@Blocking and @Incoming should support return types of CompletionStage and Uni

I tried creating a method that persists an entity upon receiving a Kafka message. 

```
@Incoming("input-topic")
@Blocking
public void processPrinterEvent(final Message<PrinterMessage> message) { 
 // create and persist entity
}
```

When building I get the following exception:
```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.5.1.Final:build (default) on project seq-pmd-backend: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.smallrye.reactivemessaging.deployment.SmallRyeReactiveMessagingProcessor#build threw an exception: javax.enterprise.inject.spi.DefinitionException: Invalid method annotated with @Incoming: com.bmw.otdlip.sepia.pmd.sequencestream.integration.kafka.boundary.PrinterTopicConsumer#processPrinterEvent - The signature is not supported as it requires 'blocking' acknowledgment, return a CompletionStage<Message<?> instead.
[ERROR]         at io.smallrye.reactive.messaging.MediatorConfigurationSupport.getIncomingError(MediatorConfigurationSupport.java:548)
[ERROR]         at io.smallrye.reactive.messaging.MediatorConfigurationSupport.validateSubscriber(MediatorConfigurationSupport.java:172)
[ERROR]         at io.smallrye.reactive.messaging.MediatorConfigurationSupport.validate(MediatorConfigurationSupport.java:84)
[ERROR]         at io.quarkus.smallrye.reactivemessaging.deployment.QuarkusMediatorConfigurationUtil.create(QuarkusMediatorConfigurationUtil.java:80)
[ERROR]         at io.quarkus.smallrye.reactivemessaging.deployment.SmallRyeReactiveMessagingProcessor.build(SmallRyeReactiveMessagingProcessor.java:315)
[ERROR]         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:932)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR]         at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
[ERROR]         at java.lang.Thread.run(Thread.java:748)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:479)
```

If I try to return a CompletionStage, as suggested by the exception, I get a different validation failing:
```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.5.1.Final:build (default) on project seq-pmd-backend: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.smallrye.reactivemessaging.deployment.SmallRyeReactiveMessagingProcessor#build threw an exception: javax.enterprise.inject.spi.DefinitionException: Invalid method annotated with @Blocking: com.bmw.otdlip.sepia.pmd.sequencestream.integration.kafka.boundary.PrinterTopicConsumer#processPrinterEvent - The @Blocking annotation is only supported for methods returning an individual Message or a payload
[ERROR]         at io.smallrye.reactive.messaging.MediatorConfigurationSupport.getBlockingError(MediatorConfigurationSupport.java:540)
[ERROR]         at io.smallrye.reactive.messaging.MediatorConfigurationSupport.validateBlocking(MediatorConfigurationSupport.java:529)
[ERROR]         at io.quarkus.smallrye.reactivemessaging.deployment.QuarkusMediatorConfigurationUtil.create(QuarkusMediatorConfigurationUtil.java:122)
[ERROR]         at io.quarkus.smallrye.reactivemessaging.deployment.SmallRyeReactiveMessagingProcessor.build(SmallRyeReactiveMessagingProcessor.java:315)
[ERROR]         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:932)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR]         at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
[ERROR]         at java.lang.Thread.run(Thread.java:748)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] -> [Help 1]
```

It looks like it's not allowed to have both `@Blocking` and `@Incoming` annotations on a method that receives a Message<>.
The reason I need this is because the topic I'm consuming messages from is a compacted topic, and in case of a tombstone message (null payload) I want to delete the entity based on the message key.

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -48,8 +48,8 @@
         <smallrye-jwt.version>2.4.1</smallrye-jwt.version>
         <smallrye-context-propagation.version>1.0.19</smallrye-context-propagation.version>
         <smallrye-reactive-streams-operators.version>1.0.13</smallrye-reactive-streams-operators.version>
-        <smallrye-converter-api.version>1.2.2</smallrye-converter-api.version>
-        <smallrye-reactive-messaging.version>2.6.1</smallrye-reactive-messaging.version>
+        <smallrye-converter-api.version>1.3.0</smallrye-converter-api.version>
+        <smallrye-reactive-messaging.version>2.7.0</smallrye-reactive-messaging.version>
         <jakarta.activation.version>1.2.1</jakarta.activation.version>
         <jakarta.annotation-api.version>1.3.5</jakarta.annotation-api.version>
         <jakarta.el-impl.version>3.0.3</jakarta.el-impl.version>
@@ -135,9 +135,9 @@
         <netty.version>4.1.49.Final</netty.version>
         <reactive-streams.version>1.0.3</reactive-streams.version>
         <jboss-logging.version>3.4.1.Final</jboss-logging.version>
-        <mutiny.version>0.11.0</mutiny.version>
-        <axle-client.version>1.2.2</axle-client.version>
-        <mutiny-client.version>1.2.2</mutiny-client.version>
+        <mutiny.version>0.12.5</mutiny.version>
+        <axle-client.version>1.3.0</axle-client.version>
+        <mutiny-client.version>1.3.0</mutiny-client.version>
         <kafka2.version>2.5.0</kafka2.version>
         <zookeeper.version>3.5.7</zookeeper.version>
         <!-- Scala is used by Kafka so we need to choose a compatible version -->

--- a/independent-projects/qute/pom.xml
+++ b/independent-projects/qute/pom.xml
@@ -40,7 +40,7 @@
         <version.jboss-logging>3.3.2.Final</version.jboss-logging>
         <version.surefire.plugin>3.0.0-M5</version.surefire.plugin>
         <version.nexus-staging-maven-plugin>1.6.8</version.nexus-staging-maven-plugin>
-        <version.smallrye-mutiny>0.11.0</version.smallrye-mutiny>
+        <version.smallrye-mutiny>0.12.5</version.smallrye-mutiny>
     </properties>
 
     <modules>

--- a/independent-projects/resteasy-reactive/pom.xml
+++ b/independent-projects/resteasy-reactive/pom.xml
@@ -49,7 +49,7 @@
         <nexus-staging-maven-plugin.version>1.6.8</nexus-staging-maven-plugin.version>
         <jboss-jaxrs-api_2.1_spec.version>2.0.1.Final</jboss-jaxrs-api_2.1_spec.version>
         <jakarta.json.version>1.1.6</jakarta.json.version>
-        <mutiny.version>0.10.1</mutiny.version>
+        <mutiny.version>0.12.5</mutiny.version>
         <smallrye-common-annotation.version>1.5.0</smallrye-common-annotation.version>
         <vertx.version>3.9.5</vertx.version>
         <rest-assured.version>4.3.3</rest-assured.version>

